src/App.js

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from 'react';
+import { BrowserRouter as Router, Route, Link } from 'react-router-dom'
+import './App.css';
+import Transactions from './components/Transactions'
+import Operations from './components/Operations'
+import Breakdown from './components/Breakdown'
+import axios from 'axios'
+
+class App extends Component {
+  constructor() {
+    super()
+    this.state = {
+      transactions: [],
+      categories: []
+    }
+  }
+
+  updateTransactionsFromDB = async () => {
+    try {
+      const tempTransactions = (await axios.get(`http://localhost:4200/transactions`)).data
+      this.setState({ transactions: tempTransactions })
+      this.updateCategories()
+    } catch (err) {
+      console.log(err.message)
+    }
+  }
+
+  updateCategories = () => {
+    let tempCategories = this.state.transactions.length && [...new Set(this.state.transactions.map(t => t.category))]
+    this.setState({categories: tempCategories})
+  }
+
+  reducer = (accumulator, currentValue) => accumulator + currentValue
+
+  getTotal = (transactions) => {
+    return transactions.length && transactions.map(t => t.amount).reduce(this.reducer)
+  }
+
+  addTransaction = async (transaction, type) => {
+    if (type === "withdrawal" && transaction.amount > 0) {
+      transaction.amount = transaction.amount * -1
+    }
+    await axios.post(`http://localhost:4200/transaction`, transaction)
+    await this.updateTransactionsFromDB()
+  }
+
+  deleteTransaction = async (id) => {
+    await axios.delete(`http://localhost:4200/transaction/${id}`)
+    await this.updateTransactionsFromDB()
+  }
+
+  render() {
+    return (
```

Comment:
I would suggest to pre define your variables here so it will be easier to read

const {transactions, categories} = this.state
const {addTransaction, updateTransactionsFrom DB, getTotal, deletTransaction} = this